### ISSUE TITLE
Add Instagram hack?

### ISSUE THREAD

**Issue by raae** [raae/gatsby-remark-oembed#128]
Should we create a hack for Instagram that skips using their oEmbed endpoint?

It would consist of getting the post's id using the schema from the provider list, and then just fill in the html that would have been returned by the oEmbed endpoint.

**Comment by nickytonline** [comment]
`hack`, sounds like a hack.😄

For the Instagram options, if there is no access token present, we can skip the oEmbed endpoint with perhaps a warning that no token was provided or some kind of message like that.

This way, it doesn't cause any friction for folks currently using the embed and if they provide a token, then that's good too.

Another idea could be an option called `useAccessToken` and the default value is set to `true` so to opt out, they need to explicitly set it to `false`.

**Comment by nickytonline** [comment]
Also, now I know why my embeds for IG aren't working. 😄

**Comment by raae** [comment]
The solution I have so far in #132 is that it skips fetching if there is no access token, and prints an error to the console. But it does not stop the build.

The PL needs a look and if you like it then it can be merged and closed.

**Comment by raae** [comment]
I agree a hack will just bloat the plugin, there is now a problem with Twitch oEmbeds as well.

**Comment by mayanksharma1133** [comment]
how to hack Instagram id 
link

**Comment by raae** [comment]
```
 {
            resolve: `@raae/gatsby-remark-oembed`,
            options: {
              providers: {
                settings: {
                  Instagram: 
                    access_token: process.env.INSTAGRAM_ACCESS_TOKEN,
                  },
                },
              },
            },
          },
```
